### ISSUE TITLE
Install to other drive

### ISSUE THREAD

**Issue by shakyevan444** [AsahiLinux/docs#69]
i boot and use an external ssd thats running macos and i want to run linux on that separate drive on a partition cuz my internal ssd is full, when i run the `curl https://alx.sh | sh` cmd it it says something about how it needs to be run in recovery or on the internal drive, recovery mode says the same thing, so when i ran off the internal drive it says it cant cuz its full, cant i just store the data on the ssd and install to the other partition??

**Comment by CodeRuner5235** [comment]
From what I understand about the boot process and permissions given to oses booted from external drives, you cant install the boot loader required to boot m1n1 -> U-boot -> grub on an enternal drive. The way the SoC boots the system (correct me if im wrong) is by loading iBoot1 off a internal SoC chip, which loads iBoot2 off of the internal nvme drive only, which then loads a macos kernal only off of the internal drive or an external drive. iBoot2 is stored on the macos recovery partiton (from what i understand) and therefore the system cannot boot without it (the recovery partition). In fact, the “boot menu” you get when you hold the power button is a macos recovery application. The way asashi linux boots is my adding a few new partitions to the internal drive for m1n1 (the macos kernal stand in which redirects iBoot2 to U-boot) U-boot (the thing that makes an uefi like enviorment for other uefi oses, like linux) then U-boot to whatever uefi boot loader for whatever os. 

Long story short, the way apple silicon boots basicaly requires that the install (of at least the m1n1 and U-Boot) happen on the internal drive or else it will strait up not work at all. So your best corse of action would be to try and move some stuff over to the external drive or get space some other way.

(to those who know more about this than me please sanity check me)

**Comment by CodeRuner5235** [comment]
Bow what you might could do is intall m1n1, U-boot, then grub on the internal drive (shouldnt take more than a gig) and have grub load linux from the external drive. That should work just fine. I have no idea how to do it though so someone smarter than me is gonna have to tell you.

**Comment by chadmed** [comment]
Comments above are the solution.
